### PR TITLE
Post message

### PR DESCRIPTION
**Pull request by rajwanialiya**
- posts page for sending a message
- stores message in fireStore db called `messages`
- error handling:
  - messageType required
  - message body required
  - at least one group (age or eligiblity group) must be selected
  - only one of province or postal code (3 char string) may be inputted
  - up to 10 postal code (3 char string) can be included

### CODE REVIEW

File: src/components/PostsSection.js [hunk 1]
```diff
@@ -1,21 +1,89 @@
...
+          selectedEligibilityGroups.push(group.id);
+        }
+      })
+
+      data.selectedAgeGroups = selectedAgeGroups;
+      data.eligibilityGroups = selectedEligibilityGroups;  
+      data.province === "All" ? data.province = "CA" : data.province = "";
+
+      if (auth.user.admin) {
+        auth.postMessage(data);
```
Comment (JJwilkin) [inline src/components/PostsSection.js:83]: we should add a try/catch here and await the auth.postMessage() call. if it doesn't throw an error setPending(false) and show a confirmation message

File: src/components/PostsSection.js [hunk 2]
```diff
@@ -1,21 +1,89 @@
...
+      for (let i = 0; i < allCheckBoxes.length; i++) {
+        allCheckBoxes[i].addEventListener("click", function() {
+          setGroupError(!this.checked)
+        });
+      }
+    }
+
+    data.postal = [];
+    allPostalCodes.forEach((postal) => {
+      if (!data.postal.includes(postal)) {
```
Comment (JJwilkin) [inline src/components/PostsSection.js:39]: if they added an empty postal code we should filter that out (if (postal && !data.postal.inclu...)

File: src/components/PostsSection.js [hunk 3]
```diff
@@ -1,21 +1,89 @@
...
+        data.postal.push(postal.value.toUpperCase())
+      }
+    })
+
+    if (province.value && province.value != "--" && data.postal.length > 0 && data.postal[0] != "") {
+      setAreaError(true);
+
+      for (let i = 0; i < allPostalCodes.length; i++) {                                                                    
+        allPostalCodes[i].addEventListener("change", function(postal) {
```
Comment (JJwilkin) [inline src/components/PostsSection.js:48]: we should look into a helper func that can be implemented on cleanup to remove all event listeners. Probably good for a future pr (https://gist.github.com/minhnc/2333095)

**Review by JJwilkin — CHANGES_REQUESTED** [review]

File: src/components/PostsSection.js (hunk 1 above)
Comment (JJwilkin) [reply src/components/PostsSection.js:83]: also, clear the form or show a success page?

Comment (rajwanialiya) [reply src/components/PostsSection.js:83]: A message will be shown to the users when the Twilio side is connected.

File: src/components/PostsSection.js (hunk 3 above)
Comment (rajwanialiya) [reply src/components/PostsSection.js:48]: Maybe add this to the tickets on the github board as something to implement in the future!

**Review by JJwilkin — APPROVED** [review]

File: src/components/FormField.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ function FormField(props) {
             ref={inputRef}
             {...inputProps}
           >
-            <option disabled selected value={defaultValue}>{defaultValue}</option>
```
Comment (aashnisshah) [inline src/components/FormField.js:21]: Did you mean to remove the disabled selected here?

File: src/components/formConstants.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+
+const ageGroups = ["18-49", "50-59", "60-79", "80+"]
```
Comment (aashnisshah) [inline src/components/formConstants.js:2]: just an FYI: I'll change the age groups here and break 18-49 into 18-29, 29-39, 40-49 to make it more flexible
I'll do that in my PR so you don't need to worry about it

File: src/components/formConstants.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+const eligibilityGroups = ["Congregate living for seniors", "Health care workers", "Adults in First Nations, Métis and Inuit populations", "Adult chronic home care recipients", "High-risk congregate settings", "Individuals with high-risk chronic conditions and their caregivers"]
```
Comment (aashnisshah) [inline src/components/formConstants.js:3]: we should store these in the backend as short form codes instead of strings

**Review by aashnisshah — APPROVED** [review]

**Pull request merged**